editor.tsx

```diff
@@ -33,8 +36,51 @@ export async function updateComponentProps(
 
     // TODO: Validate components props on schema
 
+    const loaders: PageLoaderData[] = [];
+
+    for (const component of components) {
+      const componentSchema =
+        LiveContext.getManifest().schemas[component.component];
+
+      if (!componentSchema) {
+        continue;
+      }
+
+      const componentRequiresLoader = Object.entries(
+        componentSchema.properties ?? {},
+      )
+        .filter(([, value]) => Boolean((value as JSONSchema7).$ref));
+
+      if (componentRequiresLoader.length === 0) {
+        continue;
+      }
+
+      componentRequiresLoader.forEach(([property, propertySchema]) => {
+        if (typeof propertySchema !== "object") {
+          return;
+        }
+
+        if (!propertySchema?.$ref) {
+          throw new Error(`Property ${property} doesn't have $ref on schema`);
+        }
+
+        const loaderProp = component.props[property];
```

Comment:
I don't understand this part. `loaderProp` is the current value for the component's prop that map to that loader?

Are you copying props from the component to the loaders, is that right?